libs/editor/egui_editor/src/editor.rs

```diff
@@ -253,12 +262,59 @@ impl Editor {
             ui.scroll_to_rect(self.buffer.current.cursor.end_rect(&self.galleys), None);
         }
 
+        // determine cursor markup location
+        let mut cursor_in_heading = false;
+        let mut cursor_in_bullet_list = false;
+        let mut cursor_in_number_list = false;
+        let mut cursor_in_todo_list = false;
+        let mut cursor_in_bold = false;
+        let mut cursor_in_italic = false;
+        let mut cursor_in_inline_code = false;
+
+        let ast_node_idx = self
+            .ast
+            .ast_node_at_char(self.buffer.current.cursor.selection.start());
+        let ast_node = &self.ast.nodes[ast_node_idx];
+
+        match ast_node.element {
```

Comment:
If you want this to indicate that you're in a heading when you're in a bold heading, you'll likely need to look at styled chunks which have a list of styles associated with text ranges. I'm down to merge this how it is and I'll implement that after some cleanup

Reply:
Not exactly sure how that would be done, so i'm fine merging this as what it is, and you can implement it after some cleanup.